GoogleForCreators/web-stories-wp
# Canvas: Canvas as a focusable group

## Context

Keyboard accessibility

## Summary

Add a focusable group around the canvas. This removes the canvas and the elements from the normal keyboard flow, but a user may enter into it to navigate it with keyboard.

Thanks to @BrookeGraham for pairing with me on this!

## Relevant Technical Choices

1. #11175 implemented focus group architecture. Used that for focusing the canvas vs focusing the editor
2. Added a `FocusContainer` rather than using the existing `<section />`. The reason being that the focus border was getting covered by the header. The new `FocusContainer` shows the entire focus border without having to worry about the header bar.

## To-do

n/a

## User-facing changes

![image](https://user-images.githubusercontent.com/22185279/162063407-df187d16-fefe-4a0a-b293-ad0631d48f96.png)

## Testing Instructions




This PR can be tested by following these steps:

1. Go to the editor
3. Tab over to the canvas. Should see a new container focused
4. Tabbing from there should take you to the `Previous Page` button
5. While focused on the canvas, pressing `Enter` will move focus to the background element (won't see it selected until #11000 is implemented).
6. Should be able to tab around cyclically (forward and back)


## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #11001
